lib/publiccloud/utils.pm

```diff
@@ -212,8 +212,9 @@ sub get_credentials {
     my $ua = Mojo::UserAgent->new;
     $ua->insecure(1);
     my $tx = $ua->get($url_auth);
-    die("Fetching CSP credentials failed: " . $tx->result->message) unless eval { $tx->result->is_success };
-    my $data_structure = $tx->res->json;
+    my $res = $tx->result;
```

Comment:
But I need `$res->is_success` and also `$res->json`.

If you excuse me I'd like to 'play it safe' and [follow the docummentation](https://docs.mojolicious.org/Mojo/Transaction#result).

Reply:
Using curly brackets as suggested makes no sense here and considering you use `json` later it is actually the best practice to store the result in a variable (instead of doing repeated calls to accessors). So just keep your code :-)